### DISCUSSION TITLE
Is VSG a completely a new project?

### DISCUSSION THREAD

**Discussion by IndieDev99** [vsg-dev/VulkanSceneGraph#1039] [General]
It is under MIT and don't have any OSG base under GPL or LGPL etc. (I don't have any good knowledge about that). 

I'm new to software development and game development. 
I want to learn C++ and want my own game engine based on vulkan with BSD or MIT. 

Is it good for me ? 
Any suggestions or experience. Please guide me. 
I don't want to use game engines like Godot or unity. 

Thanks in advance 😊😊

**Comment by robertosfield** [comment]
The first paragraph of the project README.md says clearly that it's MIT:

https://github.com/vsg-dev/VulkanSceneGraph#readme

The github information about the project also says clearly it's MIT.  All the headers also say clearly that it's MIT.  All the related projects like vsgXchange, vsgImGui etc. are MIT as well.

There is no OSG code, or GPL code i the VSG   The VSG is a successor to the OSG, but is a completely new API that shares a few ideas but no code.

If you plan to use C++ and Vulkan then using the VSG should me your life easier, but I wouldn't ye describe it as "easy" for someone new to programming.

Learning CMake, C++, Vulkan and VulkanSceneGraph all together is big undertaking, it's very much diving into the deep end.  I would recommend reading what I've written in the vsgTutorial online book so far.  Also looking at the examples in vsgExamples.  The forum also contains thread threads on various topics that may help.

--

What type of games/game engine do you want to develop?  What prior programming experience do you have?

**Reply by IndieDev99** [reply]
First of all, Thank You very much for VSG, and your instant reply. 

I am planning to make a low poly game like arma 2. All this is to learn C++ and game programming. I have C# expericen with Unity. And also I have some C++ experience. But still i'm learning.

**Reply by IndieDev99** [reply]
@robertosfield  one last thing. is it production ready like OSG ?

**Reply by robertosfield** [reply]
The VSG is already used in commercial applications, so for those companies it's "production ready".

However, the VSG, as a 5 year old, is still in it's infancy compared to a project like the OSG that is well over 2 decades old.  The VSG has continued to rapidly evolve since the 1.0 release this time last year.

As the VSG matures it gets used by more different types of applications, each new type pushes it in different ways, where issues are found we refactor features, or add new features to the core VSG projects or sibling projects like vsgXchange, vsgImGui etc.  

Compared to the likes of Unity you will probably find the VSG quite low level, but much more flexible so it's a tradeoff.  The VSG would be a component of a Game Engine or a simulator Image Generator, rather than one of these high level toolkits.

One of advantages that the VSG has is that in it's DNA is all the lessons leaned in developing and maintaining the OSG, it's builds on the strong points of the OSG and avoids it's pitfalls, re-imagining the scene graph for the current generation.  

Both the OSG and VSG come from a professional graphics application background rather than a game centric one, so features like handling precision requirements of GIS applications and terrestrial and space simulators, database paging etc. are integral to how they work.

We'll be releasing more companion projects that build upon the VSG over the coming months, these could well be interest to use a pat of Game Engine project that wishes to pull together various components to provide it;s functionality.

**Comment by timoore** [comment]
As a side node, this kind of "rag chewing" has its place, but maybe not in the repo discussion group. Someone brought up a Discord server a little while ago, and discussions like this might be better carried out in such a venue.

**Reply by robertosfield** [reply]
I think it's fine.  Better than trying have discussions spill over to multiple different venues.  I am busy enough just trying to answer questions on this forum and issue tracker, let along spreading myself even thinner.  As with many projects efficient use of the main developers is critical to success.

**Comment by timoore** [comment]
> ...As with many projects efficient use of the main developers is critical to success.

I think that's my point :sunglasses:

**Reply by robertosfield** [reply]
High quality support requires knowledge, perhaps in 5 years time there will be a big enough community with enough independent expertise on the VSG/Vulkan that we can man multiple different avenues and know that sound advice is being given.  I believe we some way short of that point.
